eventgrid/data-plane/src/main/java/com/microsoft/azure/eventgrid/customization/SystemEventTypeMappings.java

```diff
@@ -0,0 +1,95 @@
+
+import com.microsoft.azure.eventgrid.models.ContainerRegistryImageDeletedEventData;
+import com.microsoft.azure.eventgrid.models.ContainerRegistryImagePushedEventData;
+import com.microsoft.azure.eventgrid.models.EventHubCaptureFileCreatedEventData;
+import com.microsoft.azure.eventgrid.models.IotHubDeviceCreatedEventData;
+import com.microsoft.azure.eventgrid.models.IotHubDeviceDeletedEventData;
+import com.microsoft.azure.eventgrid.models.ResourceDeleteCancelData;
+import com.microsoft.azure.eventgrid.models.ResourceDeleteFailureData;
+import com.microsoft.azure.eventgrid.models.ResourceDeleteSuccessData;
+import com.microsoft.azure.eventgrid.models.ResourceWriteCancelData;
+import com.microsoft.azure.eventgrid.models.ResourceWriteFailureData;
+import com.microsoft.azure.eventgrid.models.ResourceWriteSuccessData;
+import com.microsoft.azure.eventgrid.models.ServiceBusActiveMessagesAvailableWithNoListenersEventData;
+import com.microsoft.azure.eventgrid.models.ServiceBusDeadletterMessagesAvailableWithNoListenersEventData;
+import com.microsoft.azure.eventgrid.models.StorageBlobCreatedEventData;
+import com.microsoft.azure.eventgrid.models.StorageBlobDeletedEventData;
+import com.microsoft.azure.management.apigeneration.Beta;
+
+import java.lang.reflect.Type;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Mapping of system event type name to corresponding type of the Java model.
+ */
+@Beta
+final class SystemEventTypeMappings {
+    /**
+     * The map containing system eventType to Java model type mapping.
+     */
+    private static Map<String, Type> systemEventMappings;
+
+    static {
+        systemEventMappings = new HashMap<>(); // key: eventType, Value:eventDataType
+        // ContainerRegistry events
+        systemEventMappings.put(SystemEventTypes.CONTAINER_REGISTRY_IMAGE_PUSHED_EVENT.toLowerCase(), ContainerRegistryImagePushedEventData.class);
+        systemEventMappings.put(SystemEventTypes.CONTAINER_REGISTRY_IMAGE_DELETED_EVENT.toLowerCase(), ContainerRegistryImageDeletedEventData.class);
+        // Device events
+        systemEventMappings.put(SystemEventTypes.IOT_HUB_DEVICE_CREATED_EVENT.toLowerCase(), IotHubDeviceCreatedEventData.class);
+        systemEventMappings.put(SystemEventTypes.IOT_HUB_DEVICE_DELETED_EVENT.toLowerCase(), IotHubDeviceDeletedEventData.class);
+        // EventGrid events
+        // TODO: Enable this once SDK is refreshed
+        // systemEventMappings.put(SystemEventTypes.EVENT_GRID_SUBSCRIPTION_VALIDATION_EVENT.toLowerCase(), SubscriptionValidationEventData.class);
+        // systemEventMappings.put(SystemEventTypes.EVENT_GRID_SUBSCRIPTION_DELETED_EVENT.toLowerCase(), SubscriptionDeletedEventData.class);
+        // Event Hub Events
+        systemEventMappings.put(SystemEventTypes.EVENT_HUB_CAPTURE_FILE_CREATED_EVENT.toLowerCase(), EventHubCaptureFileCreatedEventData.class);
+        // Media Services events
+        // TODO: Enable this once SDK is refreshed
```

Comment:
Once PR https://github.com/Azure/azure-sdk-for-java/pull/2341 is merged, hopefully that should unblock removing these commented lines.